demos/enron-demo.py

```diff
@@ -0,0 +1,75 @@
+import json
+import os
+
+import palimpzest as pz
+from palimpzest.core.lib.schemas import TextFile
+
+
+class EnronReader(pz.DataReader):
+    def __init__(self, dir: str, labels_file: str | None = None, split: str = "test"):
+        super().__init__(TextFile)
+        self.filepaths = [os.path.join(dir, filename) for filename in os.listdir(dir)]
+        self.filepaths = self.filepaths[:50] if split == "train" else self.filepaths[50:150]
+        self.filename_to_labels = {}
+        if labels_file:
+            with open(labels_file) as f:
+                self.filename_to_labels = json.load(f)
+
+    def __len__(self):
+        return len(self.filepaths)
+
+    def __getitem__(self, idx: int):
+        # get input fields
+        filepath = self.filepaths[idx]
+        filename = os.path.basename(filepath)
+        with open(filepath) as f:
+            contents = f.read()
+        
+        # create item with fields
+        item = {"fields": {}, "labels": {}}
+        item["fields"]["filename"] = filename
+        item["fields"]["contents"] = contents
+        item["labels"] = self.filename_to_labels.get(filename, {})
+
+        return item
+
+if __name__ == "__main__":
+    # create validation data source
+    val_datareader = EnronReader(dir="testdata/enron-eval-medium", labels_file="enron-eval-medium-labels.json", split="train")
+    datareader = EnronReader(dir="testdata/enron-eval-medium", split="test")
+
+    # construct plan
+    plan = pz.Dataset(datareader)
+    plan = plan.sem_add_columns([
+        {"name": "subject", "type": str, "desc": "The subject of the email"},
+        {"name": "sender", "type": str, "desc": "The email address of the email's sender"},
+    ])
+    plan = plan.sem_filter('The email refers to a fraudulent scheme (i.e., "Raptor", "Deathstar", "Chewco", and/or "Fat Boy")')
+    plan = plan.sem_filter("The email is not quoting from a news article or an article written by someone outside of Enron")
+
+    # execute pz plan
+    config = pz.QueryProcessorConfig(
+        policy=pz.MaxQuality(),
+        val_datasource=val_datareader,
+        execution_strategy="parallel",
+        max_workers=20,
+    )
+    output = plan.run(config=config, k=5, j=3, sample_budget=50)
```

Comment:
Should we add k, j, and sample_budget to Config with default values?

Reply:
I'm on the fence about this and could go either way. On the one hand, we have a global config that just has everything in it, so what's the harm in adding more things. On the other hand, these are very specific to MAB sentinel execution.